ChemMaster Update

Deantwo opened the pull request:
This kinda goes hand in hand with https://github.com/tgstation/-tg-station/pull/7403, but it is more of a refactoring/update~~/nerf~~ so I left it out of that PR.

**ChemMaster changes:**
- Analyze refactored for readability.
- Pill bottles can no longer be inserted into a CondiMaster.
- Main UI updated to the station standard along with some nice changes.
- Can now dispense bottles/condiments without inserting a beaker.
- ~~Glass is now used when creating bottles. Glass can be refilled with
  sheets of glass. Can store enough for glass for 40 bottles or 16
  condiments, but starts with only half that.~~
- Custom amount transfers now have a cancel button too.
- Changed "Create Pills" to "Create multiple pills" and moved the "(max 10)" to the input window.
- Many minor code changes. (Line spaces, readability, removed colons, etc.)
- Fixes https://github.com/tgstation/-tg-station/issues/7091.

![chemmaster](https://cloud.githubusercontent.com/assets/2676134/6059403/43e5ae40-ad31-11e4-8169-0bd295971432.PNG)

~~**Other changes:**~~
- ~~Added "g_amt = 200" to bottle.~~
- ~~Added "g_amt = 500" to condiment.~~

**Feedback needed for:**
- ~~Glass storage size.~~
- ~~Glass storage start amount.~~
- ~~Bottle glass cost.~~
- Interface look and feel.

Deantwo committed:
ChemMaster UI Update

- Analyze refactored for readability.
- Pill bottles can no longer be inserted into a CondiMaster.
- Main UI updated to the station standard along with some nice changes.

Deantwo committed:
Merge branch 'GloomyChemMaster' into ChemMasterGetsGloomy

Deantwo committed:
ChemMaster  UI Update 2

- Can now disspense bottles/condiments without inserting a beaker.
- Glass is now used when creating bottles. Glass can be refilled with
sheets of glass. Can store enough for glass for 40 bottles or 16
condiments, but starts with only half that.
- Custom amount transfers now have a cancel button too.
- Many many minor changes.

Other:
- Added "g_amt = 200" to bottle.
- Added "g_amt = 500" to condiment.

Deantwo commented:
*Some numbers for math people:*
- The ChemMaster and CondiMaster can both store 8000 glass.
- The storage is only half full on round start.
- One sheet of glass is worth 2000 glass (same as Autolathe).
- That is a total of 4 sheets of glass to fill it up from empty.
- Bottles cost 200 glass.
- Resulting in (8000 / 200 =) 40 bottles from full, and 20 bottles from round start.
- Condiments cost 500 glass.
- Resulting in (8000 / 500 =) 16 bottles from full, and 8 bottles from round start.

Deantwo commented:
There is no way to get the glass out again, but since we are talking no more then 4 sheets of glass, I don't think it will matter.
IC reason could be that the glass is stored in melted/liquid form.

pudl commented:
needing sheets of glass is stupid, chemists dont have time to be doing shit a cargo tech would normally be doing

Deantwo commented:
Question is, should they ever need more than 40 bottles?
If they need that many bottles, they are doing something wrong.

Besides, they can ask the other departments if they have a sheet or two they can spare.

AlexCrimson commented:
Yeah no. Fuck that. Requiring materials for this stuff is insane.

Kromgar commented:
Muh realism

Kraseo commented:
![efecto-positivo](https://cloud.githubusercontent.com/assets/3855300/5978331/61a71a86-a89f-11e4-88e6-1b530313a388.gif)

Deantwo commented:
Before people complain about the title too.

So rather than yell "no", wanna come up with a better idea?
Or do I just comment out the glass part and be done with it?

AlexCrimson commented:
What is the reasoning behind adding that mechanic in the first place? Was there any bottle spam going on? Are bottles in need of balancing? If it a justified nerf then i would try to understand, but i can see no reasoning for it. It just seems like it will make managing Chemistry needlessly tedious.

Deantwo commented:
Ideas like "increase glass storage capacity to X" and "increase starting glass storage amount to X" would be way more useful then "no".
But that at least your second comment has more text to it.

Miauw62 commented:
why

Deantwo commented:
Main reason for thinking up this crazy idea was this: https://github.com/tgstation/-tg-station/pull/7402#issuecomment-71965520

Other than that, the changes I made to the UI allows you to create bottles without inserting a beaker first. So thought this would be a nice thing to add with it, as it would not require less brain to do said bottle spamming.

dannno commented:
"Why" is a perfectly reasonable question here

JJRcop commented:
https://github.com/tgstation/-tg-station/pull/7402#issuecomment-71965520 suggested that bottles shouldn't be able to be refilled, not this

AlexCrimson commented:
I believe that comment is talking about bottles of alcohol, not Chemistry 30unit bottles. The issue is that the large bottles of alcohol and drinking glasses allow you to see the reagents in them, so they are better than beakers. Your previous PR should fix that, no?

The idea of nerfing Chemistry bottles still confuses me. I have never seen a massive amount of bottle spam, and certainly never seen it done in a way that was overpowered or required balancing. Usually it happens when a Chemist is inefficient with their chemical mixing and ends up with extra chemicals. Hardly a scenario that requires setting limitations like this.

So i say again... fuck no.

Deantwo commented:
Because infinite stuff is bad?

Ok then I'll just comment out the glass part.
._.

Miauw62 commented:
if anything, infinite bottles are good because they give you some expendable trash that isnt just empty food.

Deantwo commented:
And the disposal crusher thingy doesn't get to spit out glass when nomming on bottles.
At least I don't think it currently does. Let me check.

Miauw62 commented:
AFAIK it spits out glass and metal randomly?

Deantwo commented:
Wait what? It doesn't check what it cost to make the item it crushes and spit out (most of) that?

AlexCrimson commented:
I thought it had a random chance to generate metal/glass/plasteel for each item it crushed. Atleast thats what ive always been told.

Deantwo commented:
I tossed in 30ish bottles and it spit out 9 metal sheets and 2 glass sheets.
So yeah you are right. Weird way to do it, but I guess it works.
Kinda like a slot machine now, that cost me about 3 glass sheets (with this change) and I won 2.

Deantwo committed:
Reverted Finite Bottles

Also removed "g_amt" from bottle.dm and condiment.dm again.

Deantwo commented on code/modules/reagents/Chemistry-Machinery.dm:
```diff
@@ -277,15 +278,14 @@
 
 
-/obj/machinery/chem_master/attackby(var/obj/item/weapon/B as obj, var/mob/user as mob)
+/obj/machinery/chem_master/attackby(var/obj/item/B as obj, var/mob/user as mob)
```
Forgot to revert this line when reverting finite bottles, anyone think it'll mater?

Miauw62 replied:
no

Deantwo commented:
Anyway, there you go.
Pretty new UI for the ChemMaster with all the bottles your heart may desire.

Deantwo committed:
Changelog

This I think needs a changelog.

Tenebrosity commented:
Does the uh does the condiment master label condiment bottles in this pull?

Deantwo commented:
It didn't before, so I didn't change it.
Condiments automaticly change name depending on what you put in them seemingly.

Deantwo commented:
Why a "Feature" tag?
Any feature (aka finite bottles) have been reverted.

pudl commented:
>Can now dispense bottles/condiments without inserting a beaker.

Deantwo commented:
Only reason for that change is that the whole UI isn't gone/hidden before you insert a beaker.
You could always dispense bottles/condiments without adding anything to the buffer.

Deantwo commented:
This can be merged.
Unless someone can see any errors or something else that need changed on the ChemMaster UI.

Deantwo commented:
Mh, I should add a fix for #7091 before merging this I guess, wouldn't need much and I already touched that pard of the code anyway.
**Closed until done.**

Deantwo committed:
Late Fixes

- Removed spaces. (such a huge nerf!)
- Fixes https://github.com/tgstation/-tg-station/issues/7091
- Fixes two little errors I made on the CondiMaster's UI.
- Removes "No pill bottle inserted." from CondiMaster.
- Changes the window's default size a little to better suit the changes.
